homework-submission/myapp/src/Components/Data.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import Todos from './Todos.json';
+import Form from './Form';
+
+
+class Data extends React.Component {
+  state = {
+    Todos
+  };
+
+  descriptionInput = React.createRef();
+  deadlineInput = React.createRef();
+
+  addTodo = event => {
+    event.preventDefault();
+
+    const descriptionInput = this.descriptionInput.current.value;
+    const deadlineInput = this.deadlineInput.current.value;
+    const newState = this.state;
+
+    const Todos = newState.Todos;
+    Todos.push({
+      id: Todos.length + 1,
```

Comment:
What is the `id` for the next one of this if this `todo` has been deleted?

Well in your case you are re-rendering the whole thing, which is fine.

But in a different situation you will face a Bug may be hard to follow or fix, so may have to re-think about how do I specify an `Id` for a new Item.